src/map/storage.c

```diff
@@ -180,29 +182,35 @@ int storage_additem(struct map_session_data* sd, struct item* item_data, int amo
 	{//Stackable
 		for( i = 0; i < MAX_STORAGE; i++ )
 		{
-			if( compare_item(&stor->items[i], item_data) )
+			it = &VECTOR_INDEX(sd->storage, i);
+			if( compare_item(it, item_data) )
 			{// existing items found, stack them
-				if( amount > MAX_AMOUNT - stor->items[i].amount || ( data->stack.storage && amount > data->stack.amount - stor->items[i].amount ) )
+				if( amount > MAX_AMOUNT - it->amount || ( data->stack.storage && amount > data->stack.amount - it->amount ) )
```

Comment:
`if (amount > MAX_AMOUNT - it->amount || (data->stack.storage && amount > data->stack.amount - it->amount))`